images/win/scripts/ImageHelpers/PathHelpers.ps1

```diff
@@ -1,68 +1,56 @@
-function Test-MachinePath{
+function Get-SystemVariable {
     [CmdletBinding()]
     param(
-        [string]$PathItem
+        [string]$SystemVariable
     )
-
-    $currentPath = Get-MachinePath
-
-    $pathItems = $currentPath.Split(';')
-
-    if($pathItems.Contains($PathItem))
-    {
-        return $true
-    }
-    else
-    {
-        return $false
-    }
+    return [System.Environment]::GetEnvironmentVariable($SystemVariable, "Machine")
 }
 
-function Set-MachinePath{
+function Set-SystemVariable {
     [CmdletBinding()]
     param(
-        [string]$NewPath
+        [string]$SystemVariable,
+        [string]$Value
     )
-    Set-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name Path -Value $NewPath
-    return $NewPath
+    [System.Environment]::SetEnvironmentVariable($SystemVariable, $Value, "Machine")
+    return Get-SystemVariable $SystemVariable
 }
 
-function Add-MachinePathItem
-{
+function Get-MachinePath {
     [CmdletBinding()]
     param(
-        [string]$PathItem
-    )
 
-    $currentPath = Get-MachinePath
-    $newPath = $PathItem + ';' + $currentPath
-    return Set-MachinePath -NewPath $newPath
+    )
+    return Get-SystemVariable PATH
 }
 
-function Get-MachinePath{
+function Set-MachinePath {
     [CmdletBinding()]
     param(
-
+        [string]$NewPath
     )
-    $currentPath = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name PATH).Path
-    return $currentPath
+    Set-SystemVariable PATH $NewPath
+    return Get-SystemVariable PATH
```

Comment:
Do we need call twice `Get-SystemVariable PATH` at line 33 and in the `Set-SystemVariable` function at line 16?

Reply:
You are right. There's no need to return anything here since it's already returned in "Set-SystemVariable"

Reply:
Updated